lib/secretsscanner/reporter/report_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reporter_test
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"sort"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	"github.com/gravitational/teleport/api/defaults"
+	accessgraphsecretsv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/accessgraph/v1"
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/api/types/accessgraph"
+	dtassert "github.com/gravitational/teleport/lib/devicetrust/assert"
+	dtauthn "github.com/gravitational/teleport/lib/devicetrust/authn"
+	dttestenv "github.com/gravitational/teleport/lib/devicetrust/testenv"
+	"github.com/gravitational/teleport/lib/secretsscanner/reporter"
+)
+
+func TestReporter(t *testing.T) {
+	// disable TLS routing check for tests
+	t.Setenv(defaults.TLSRoutingConnUpgradeEnvVar, "false")
```

Comment:
Why do we need to set this? Is it for "secretsscannerclient.NewSecretsScannerServiceClient"?

Reply:
Yes, it includes a check for TLS routing, which we need to override because we don't run a full proxy API.